packages/nimble-components/src/rich-text/specs/mention-hld.md

```diff
@@ -0,0 +1,617 @@
+        text="username-3"
+        character="@"
+    ></nimble-mapping-mention-users>
+    ```
+3. The UI of the list option is completely dependent on the client application.
+4. All the keyboard and mouse interactions within the dropdown list should be handled internally within the editor as `nimble-list-option`
+   does not support any.
+
+#### Client Usage Guidance on Filtered Users:
+
+Initially, the client application need not provide any user list within the children of `nimble-rich-text-editor` and it can be empty.
+The editor component will emit an event whenever the `@` character is entered into the editor. Client
+can listen to the `mention-update` event and provide the initial user lists.
+
+It is recommended to `sort` the usernames in alphabetical order and sent at most twenty users list at a time to reduce the number of elements
+in the DOM for a single page. This dynamic loading of users list will help us specifically in comments feature when there are lot of `@mention`
+happens in a single page and if the users list is huge, it may cause the page to load slower as the mapping mention element is provided
+for every editor.
+
+The `mention-update` event will also be triggered when the user types any character after `@`, containing that text following the `@` character.
+For instance, if a user types `@` and then adds `a` the event will be emitted with data that includes the value `@a`. The client will be
+listening to this event, filter the list of users that includes the names containing the letter `a` and then dynamically update the
+`nimble-mapping-mention-user` element based on the filter data. Subsequently, a maximum of twenty filtered options should be transmitted to the
+editor.
+
+_Note_: The editor will also perform the same filtering once again to ensure the filtered options are proper and update the dropdown list in the UI.
+This helps to filter the list, regardless of whether the client is loading the list dynamically by listening to the event as mentioned above
+or statically providing user details at the start via `nimble-mapping-mention`.
+
+### API
+
+#### Events
+
+-   `mention-update` - This can be achieved through Tiptap's `onUpdate()` and `onStart()` methods in `render` function in
+    [suggestion](https://tiptap.dev/api/utilities/suggestion#render) configurations. This event fires with the `eventData` containing the
+    current text that is added after the `@` character.
+
+    This event will be triggered in the following scenarios to perform filtering in the client application:
+
+    1. When a user inserts the character (e.g., `@`) into the editor, which activates the mention popup.
+    2. When a user adds or removes text after inserting the mention character into the editor.
+    3. When a user repositions the cursor between the text segments added after the mention character.
+    4. When there is configuration element `nimble-rich-text-mention-users` but with no mapping elements.
```

Comment:
I'm confused by this one since it's not a specific user interaction. Should this be a separate condition that says that the actions in 1-3 will trigger the event as long as there is a `nimble-rich-text-mention-users` element, regardless of how many mapping elements it has?

Reply:
Yeah, this should not be a part of the other user interaction points so I have moved it to the start of explaining this.